src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +561,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
+                While CanGet(Here)
+                    ch = Peek(Here)
+                    If IsWhitespace(ch) Then
+                        Here += 1
+                    Else
+                        Exit While
+                    End If
+                End While
+                If CanGet(Here) AndAlso IsSingleQuote(Peek(Here)) Then
+                    ' If you get here you have a Line Continuation with have 0 or more spaces followed by a  comment
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
+                    Dim comment As SyntaxTrivia = ScanComment()
+                    If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        comment = DirectCast(comment.SetDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                                                                                                    New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))}), SyntaxTrivia)
+                    End If
+                    tList.Add(comment)
+                    ch = Peek()
+                    atNewLine = IsNewLine(ch)
                 Else
-                    Exit While
+                    ' We have a Line Continuation without comment but have 0 or more spaces after _, so process as V15.5
+                    If Not CanGet(Here) Then
```

Comment:
>If Not CanGet(Here) Then [](start = 20, length = 24)

It feels like this new condition will make an error a situation when '_', followed by a whitespace, followed by EOF. It doesn't look like that was an error before.
I am not sure why this code path needs any change from the original code. I would restore the original code that is based on the character peeked at the loop above, i.e.
```
            atNewLine = IsNewLine(ch)
            If Not atNewLine AndAlso CanGet(Here) Then
                Return False
            End If

            tList.Add(MakeLineContinuationTrivia(GetText(1)))
```

Reply:
Yes that restores the logic to match 15.X and I the issue of needed to change EOFToken to BadToken.